### PR TITLE
Have `ErrorProneRuntimeClasspath` ignore non-public types

### PR DESCRIPTION
**Pull request by Stephan202**
~:exclamation: This PR is on top of #976. :exclamation:~

Suggested commit message:
```
Have `ErrorProneRuntimeClasspath` ignore non-public types (#972)
```

### CODE REVIEW

**Comment by Stephan202** [comment]
Okay, the Pitest analysis now fails. Seems to be due to `junit-jupiter-engine` dependencies being dropped in #973. Two thoughts:
1. Would have been nice if CI had caught this, but that's perhaps too expensive to achieve.
2. I guess I should report this upstream, as it's counter-intuitive that Surefire doesn't require this dependency to be specified explicitly, while Pitest does.

**Comment by Stephan202** [comment]
I filed pitest/pitest-junit5-plugin#105. Let's see whether there's a fast response; I can open a PR to revert most of #973 tomorrow.

**Comment by Stephan202** [comment]
I rebased the branch on top of #976; the build should now pass :eyes:

**Review by Venorcis — APPROVED** [review]

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/util/ThirdPartyLibraryTest.java [hunk 1]
```diff
@@ -86,23 +87,64 @@ void isIntroductionAllowedIgnoreClasspathCompat(boolean ignoreClassPath) {
...
+   */
+  @BugPattern(severity = ERROR, summary = "Interacts with `ThirdPartyLibrary` for testing purposes")
+  public static final class CanIntroduceUsageTestChecker extends BugChecker
+      implements ClassTreeMatcher {
+    private static final long serialVersionUID = 1L;
+    private static final ImmutableMap<String, String> TYPES =
+        ImmutableMap.of(
+            "GUAVA_PUBLIC",
+            ImmutableList.class.getCanonicalName(),
+            "GUAVA_PRIVATE",
```
Comment (rickie) [inline error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/util/ThirdPartyLibraryTest.java:133]: Depends on how you look at it, but strictly speaking this type is not `private`, right?

**Review by rickie — APPROVED** [review]
Nice improvement! 🚀

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/util/ThirdPartyLibraryTest.java (hunk 1 above)
Comment (Stephan202) [reply error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/util/ThirdPartyLibraryTest.java:133]: Indeed, it's package-private, but here I meant private-to-the-library/implementation.

**Pull request merged**